clients/src/main/java/org/apache/kafka/common/record/CompressionType.java

```diff
@@ -179,4 +185,17 @@ private static MethodHandle findConstructor(String className, MethodType methodT
         }
     }
 
+    static class RecyclingBufferSupplier implements KafkaLZ4BlockInputStream.BufferSupplier {
```

Comment:
It's a little odd for this to live here given the interface is in `KafkaLZ4BlockInputStream`. I'd suggest either moving it there or pulling the `BufferSupplier` interface out of `KafkaLZ4BlockInputStream`.